cmd/exec/cmd.go

```diff
@@ -0,0 +1,157 @@
+// Copyright 2024 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exec
+
+import (
+	"context"
+	"fmt"
+
+	contextCMD "github.com/okteto/okteto/cmd/context"
+	"github.com/okteto/okteto/cmd/utils"
+	"github.com/okteto/okteto/pkg/analytics"
+	"github.com/okteto/okteto/pkg/log/io"
+	"github.com/okteto/okteto/pkg/model"
+	"github.com/okteto/okteto/pkg/okteto"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+// execFlags is the input of the user to exec command
+type execFlags struct {
+	manifestPath string
+	namespace    string
+	k8sContext   string
+}
+
+type onExecFinish func()
+
+type metadataTracker interface {
+	SetMetadata(metadata *analytics.TrackExecMetadata)
+	Track()
+}
+
+// Exec executes a command on the remote development container
+type Exec struct {
+	ioCtrl            *io.Controller
+	fs                afero.Fs
+	appRetriever      *appRetriever
+	k8sClientProvider okteto.K8sClientProvider
+	mixpannelTracker  metadataTracker
+
+	getExecutorFunc func(dev *model.Dev, podName string) (executor, error)
+	onExecFinish    []onExecFinish
+}
+
+// NewExec creates a new exec command
+func NewExec(fs afero.Fs, ioCtrl *io.Controller) *Exec {
+	k8sProvider := okteto.NewK8sClientProvider()
+	mixpannelTracker := &mixpannelTrack{
+		trackFunc: analytics.TrackExec,
+	}
+	e := &Exec{
+		ioCtrl:            ioCtrl,
+		fs:                fs,
+		k8sClientProvider: k8sProvider,
+		appRetriever:      newAppRetriever(ioCtrl, k8sProvider),
+		mixpannelTracker:  mixpannelTracker,
+		onExecFinish: []onExecFinish{
+			mixpannelTracker.Track,
+		},
+	}
+	e.getExecutorFunc = e.getExecutor
```

Comment:
Maybe I'm missing something, but I don't get why this? `getExecutor` is already a function of `e`, why do we need to set a field of the struct with that function?